lib/protocol/errors.js

```diff
@@ -552,15 +552,18 @@ class ProxyRequestError extends ES6Error {
     super(err || message);
 
     this.w3cStatus = HTTPStatusCodes.BAD_REQUEST;
+
+    // If a string was provided, parse the string
     if (_.isString(responseError)) {
-      this.jsonwp = JSON.parse(responseError);
+      responseError = JSON.parse(responseError);
```

Comment:
shouldn't this be enclosed in try/catch ?

Reply:
I think it's actually supposed to throws an exception if the `responseError` is malformed, that way it will be caught and returned as an `UnknownError` and show the stacktrace and the JS error message.

Reply:
ok if you say so. Is such case covered by a test?

Reply:
I don't think so. I could add one.

Reply:
Looks like it's impossible for `responseError` to be a non-object

The only place ProxyRequestError constructor is called is here:

https://github.com/appium/appium-base-driver/blob/master/lib/jsonwp-proxy/proxy.js#L144

...which does a JSON.parse on strings beforehand

I'd be fine with just taking the `JSON.parse` block out altogether, but I'm nervous it might break something.